src/server.ts

```diff
@@ -103,24 +102,20 @@ if (config.dashboard.enabled && config.dashboard.dist_path) {
   // app.set('views', clientDirectory);
 }
 
-app.get('/api/subscribe', authorize, (req: Request, res: Response) => {
+app.get('/api/subscribe', handleDebugAuth, (req: Request, res: Response) => {
   nestedCountersInstance.countEvent('api', 'subscribe')
   const query = req.query
-  if (!query || !req.ip || !query.port) {
+  if (!query || !query.ip || !query.port) {
     console.log('Invalid ip or port')
     return res.end('Invalid ip or port')
   }
-  const ip = req.ip || '127.0.0.1'
-  const port = req.connection.localPort || 9001
-  const success = changeNode(ip, port, true)
-  if (!success) {
-    res.end(`Ip not in the nodelist ${ip}:${port}, node subscription rejected`)
-    return
-  }
+  const ip = query.ip.toString() || '127.0.0.1'
+  const port = parseInt(query.port.toString()) || 9001
+  changeNode(ip, port)
```

Comment:
This function can fail and this should be handled. The validation of the input is done in changeNode. If ip is not a valid IP address then nothing happens.